pythonfmu/builder.py

```diff
@@ -201,13 +200,29 @@ def build_FMU(
         newargs: dict | None = None,
         **options,
     ) -> Path:
+        """ Build the FMU from the Python script, additional project files and documentatiion.
+
+        Args:
+            script_file (FilePath): The main Python script containing the Python model class
+            dest (FilePath)='.': Optional destination path.
+               If this is a full file name with '.fmu' extension, it is used as FMU file name.
+               Otherwise the FMU file name is constructed automatically from the script file name.
+            project_files (Iterable[FilePath]): Optional list/tuple of additional project files needed to run model
+            documentation_folder (FilePath): Optional additional documentation (beyond modelDescription)
+            newargs (dict): Optional dict of replacements of model class __init__() arguments.
+        """
         script_file = Path(script_file)
         if not script_file.exists():
             raise ValueError(f"No such file {script_file!s}")
         if not script_file.suffix.endswith(".py"):
             raise ValueError(f"File {script_file!s} must have extension '.py'!")
-
+        
         dest = Path(dest)
+        if dest.is_file() or (not dest.is_dir() and dest.suffix == '.fmu'): # explicit FMU file name is provided
```

Comment:
This boolean logic seems a bit weird, in order to have `dest.suffix == '.fmu'`, `dest.is_file()` needs to return true. From what I can see, you want to use the dest to be determine the FMU filename if this is effectively a file that ends with `.fmu` extension. As you're using OR, if the dest parameter is a file but not necessarily an FMU, then this condition is still true and not in line with the docstring you wrote. I suggest:

```suggestion
        dest = Path(dest)
        if dest.is_file() and dest.suffix == '.fmu': # explicit FMU file name is provided
```

Reply:
After discussing it seems, that my suggestion wasn't quite correct as is_file() will be false if the file doesn't exist. And this code shouldn't necessarily require the file to exist